NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -123,136 +636,382 @@ def calculate_simulation_weights(positions, zenith, azimuth, site='summit', debu
+        returns the maximum value of the electric field provided by coreas
+        """
+        if not self.star_shape_initialized:
+            logger.error('interpolator not initialized, call initialize_star_shape first')
+            return None
+        else:
+            return self.max_coreas_efield
+
+    def get_star_radius(self):
+        """
+        returns the maximal radius of the star shape pattern in the shower plane
+        """
+        if not self.star_shape_initialized:
+            logger.error('interpolator not initialized, call initialize_star_shape first')
+            return None
+        else:
+            return self.star_radius
+
+    def get_geo_star_radius(self):
+        """
+        returns the maximal radius of the star shape pattern on ground
+        """
+        if not self.star_shape_initialized:
+            logger.error('interpolator not initialized, call initialize_star_shape first')
+            return None
+        else:
+            return self.geo_star_radius
+
+    def initialize_efield_interpolator(self, interp_lowfreq, interp_highfreq):
+        """
+        Initialises the efield interpolator object. The efield will be interpolated in the shower plane for
+        geometrical reasons. If the geomagnetic angle is smaller than 15deg, no interpolator object is returned.
+
+        Parameters
+        ----------
+        interp_lowfreq : float
+            lower frequency for the bandpass filter in interpolation in GHz
+        interp_highfreq : float
+            higher frequency for the bandpass filter in interpolation in GHz
+
+        Returns
+        -------
+        efield_interpolator : interpolator object
+
+        """
+        self.efield_interpolator_initialized = True
+        self.interp_lowfreq = interp_lowfreq
+        self.interp_highfreq = interp_highfreq
+
+        geomagnetic_angle = get_geomagnetic_angle(self.zenith, self.azimuth, self.shower[shp.magnetic_field_vector])
+
+        if geomagnetic_angle < 15 * units.deg:
+            logger.warning(
+                f'geomagnetic angle is {geomagnetic_angle / units.deg:.2f} deg, '
+                f'which is smaller than 15deg, which is the lower limit for the signal interpolation. '
+                f'The closest obersever is used instead.')
+            self.efield_interpolator = -1
+        else:
+            logger.info(
+                f'electric field interpolation with lowfreq {interp_lowfreq / units.MHz} MHz '
+                f'and highfreq {interp_highfreq / units.MHz} MHz')
+
+            self.efield_interpolator = cr_pulse_interpolator.signal_interpolation_fourier.interp2d_signal(
+                self.obs_positions_vxB_vxvxB[:, 0],
+                self.obs_positions_vxB_vxvxB[:, 1],
+                self.electric_field_on_sky,
+                signals_start_times=self.efield_times[:, 0],
+                lowfreq=(interp_lowfreq - 0.01) / units.MHz,
+                highfreq=(interp_highfreq + 0.01) / units.MHz,
```

Comment:
Why the -/+ 0.01 ?

Reply:
I added a margin to avoid edge effects when applying filters. But I am not sure if it is really necessary. Feel free to delete it.